### ISSUE TITLE
How does react map props to dom internally?

### ISSUE THREAD

**Issue by lifeiscontent** [facebook/react#11706]
Hi, I'm working on a library that proxy's SVG elements so you can mutate the SVG Element when it has loaded.

I noticed that when I added an `onClick` prop to my Mutate component it didn't map properly as in javascript the proper syntax is `onclick`.

I'm wondering how ReactDOM does this internally so I can have a consistent API with my library.

any support on this topic is greatly appreciated, thanks!

**Comment by lifeiscontent** [comment]
@gaearon yeah, I understand that JSX uses camelCase, but what code in the react library does the mapping of the camelCase props to dom elements? If you look at the code I've pointed to, its working with real dom nodes, but if there is a way where I can use react instead of the dom, that'd be even better but I'm currently not sure how I could possibly do that.

**Comment by gaearon** [comment]
There is a mapping between DOM events and names but it's kind of scattered around the library right now, and there's no easy snippet to cut and paste.

**Comment by lifeiscontent** [comment]
@gaearon can you point me to a few of the files these are defined in?

**Comment by gaearon** [comment]
Here's some events that get their first letter capitalized with `on` added in front:

https://github.com/facebook/react/blob/c78db58a61e0ed5d5ba9a4b1449146137a532c27/packages/react-dom/src/events/SimpleEventPlugin.js#L55-L119

There's more you can find if you search for `top*` in the code. Some are here: https://github.com/facebook/react/blob/master/packages/react-dom/src/events/BrowserEventConstants.js

But there is more.

**Comment by lifeiscontent** [comment]
@gaearon thanks for the help, quick question. how do you inline code like that in GitHub? or where are the docs that describe that functionality?

**Comment by gaearon** [comment]
Just paste a link to specific lines and it'll be inlined. You can take a link to specific lines by clicking on the start line in the file source view. (and shift+clicking to choose the end line)
